pytest_fauxfactory.py

```diff
@@ -15,18 +20,42 @@
 )
 
 
-def gen_string(items=None, str_type=None, *args, **kwargs):
-    """Generate a new string type."""
-    item = 0
+class GenString(object):
+    """fauxfactory string generator helper"""
+    __slots__ = ['str_type', 'args', 'kwargs']
+
+    def __init__(self, str_type, *args, **kwargs):
+        if str_type is None:
+            str_type = fauxfactory.gen_choice(STRING_TYPES)
+
+        if (not isinstance(str_type, six.string_types)
```

Comment:
How about a test that does pass a string but that is not callable? It should expect a `ValueError` exception.